WIP: Reactstrap Header

vikr01 opened the pull request:
## Pull request name
<!-- Make sure that your PR is not a duplicate -->

#### Pre-Submission Checklist
<!-- Go over all points below, and after creating the PR, tick all the checkboxes that apply. -->
<!-- All points should be verified, otherwise, read the CONTRIBUTING guidelines from above-->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] Your pull request targets the `master` branch.
- [x] Branch starts with either `fix/` or `feature/` (e.g. `fix/signin-issue`)
- [x] All new and existing tests pass.

#### Type of Change
<!-- What type of change does your code introduce? After creating the PR, tick the checkboxes that apply. -->
- [ ] Small bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)

#### Checklist:
<!-- Go over all points below, and after creating the PR, tick the checkboxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] Tested changes locally.
- [ ] Closes currently open issue (replace XXXX with an issue no): Closes #XXXX

#### Description
<!-- Describe your changes in detail -->
- Header has been rewritten with ReactStrap components.
- Utilized `SyntheticEvent` type from React to fix flow errors on cases of `Event<T>`.

amilajack commented:
Nice! Can you share screenshots of all the pages that were changed?

amilajack commented on app/components/header/Header.js:
```diff
@@ -1,6 +1,17 @@
+  NavItem
+} from 'reactstrap';
 import React, { Component } from 'react';
+import type { Node, SyntheticEvent as Event } from 'react';
```
type imports should go before relative imports. relative type imports should go after relative imports

amilajack commented on app/components/header/Header.js:
```diff
@@ -31,24 +42,24 @@ export default class Header extends Component {
    * Set the mode of the movies to be 'search'
    */
-  setSearchState(searchQuery: string) {
+  setSearchState(searchQuery: string): void {
```
`: void` annotation is unnecessary

vikr01 commented:
Fixed the type imports and removed the void declarations. No pages were changed. Though, this isn't really utilizing React features -- it's just using the corresponding component from `reactstrap` instead of the html element.
